internal/core/src/expr/ITypeExpr.h

```diff
@@ -29,6 +29,62 @@
 namespace milvus {
 namespace expr {
 
+using GenericValue = proto::plan::GenericValue;
+
+struct GenericValueEqual {
+    bool
+    operator()(const GenericValue& lhs, const GenericValue& rhs) const {
+        if (lhs.val_case() != rhs.val_case())
+            return false;
+        switch (lhs.val_case()) {
+            case GenericValue::kBoolVal:
+                return lhs.bool_val() == rhs.bool_val();
+            case GenericValue::kInt64Val:
+                return lhs.int64_val() == rhs.int64_val();
+            case GenericValue::kFloatVal:
+                return lhs.float_val() == rhs.float_val();
+            case GenericValue::kStringVal:
+                return lhs.string_val() == rhs.string_val();
```

Comment:
for other not supported, throw exception

Reply:
If the data type is not supported, it should still be able to search. A guard of data type might be needed if throwing exceptions here.